Lib/http/server.py

```diff
@@ -1052,14 +1081,21 @@ def server_bind(self):
 
         def finish_request(self, request, client_address):
             self.RequestHandlerClass(request, client_address, self,
-                                     directory=args.directory)
+                                     directory=args.directory,
+                                     response_headers=self.response_headers)
```

Comment:
Why not do this the same way the _--directory_ or _--protocol_ options are implemented? Either way should avoid adding internal parameters to unrelated HTTPServer and BaseRequestHandler classes.

You could build the _response_headers_ dictionary before the DualStackServerMixin class, and then pass it by referencing the outer scope like is already done with _args.directory_:

```suggestion
                                     response_headers=response_headers)
```

Or set the _response_headers_ attribute on the SimpleHTTPRequestHandler class rather than in its constructor, like is done for _protocol_verison_ in the _test_ function.

Reply:
I followed your advice, see b1026d2a1faeec3c323467f5d7ca354f0712604a. I made `response_headers` an argument to `SimpleHTTPRequestHandler` only, and send the argument to it in the `DualStackServerMixin` class.